eos-update-flatpak-repos

```diff
@@ -1335,6 +1353,30 @@ def copy_commit(repo, src_rev, dest_ref,
     return dest_checksum, vendor_prefixes
 
 
+def migrate_extra_data(old_ref, new_ref):
+    old_deploy = old_ref.get_deploy_dir()
+    old_extra = os.path.join(old_deploy, 'files', 'extra')
+
+    new_deploy = new_ref.get_deploy_dir()
+    new_extra = os.path.join(new_deploy, 'files', 'extra')
+
+    if os.path.exists(new_extra):
+        # either the user manually installed Skype from Flathub, or we got
+        # this far but died before uninstalling the old ref last time
+        logging.info('%s already has extra data dir', new_ref.format_ref())
+    else:
+        logging.debug('Copying extra data from %s to %s',
+                      old_ref.format_ref(), new_ref.format_ref())
+        subprocess.check_call(['cp', '-al', old_extra, new_extra + '.tmp'])
```

Comment:
This behaves differently if the target directory already exists, so will not resume correctly if interrupted:

```
$ find /tmp/tmp.HldWAbt7ur/
/tmp/tmp.HldWAbt7ur/
/tmp/tmp.HldWAbt7ur/source
/tmp/tmp.HldWAbt7ur/source/b
/tmp/tmp.HldWAbt7ur/source/a
$ cp -al /tmp/tmp.HldWAbt7ur/source /tmp/tmp.HldWAbt7ur/target
$ cp -al /tmp/tmp.HldWAbt7ur/source /tmp/tmp.HldWAbt7ur/target
$ find /tmp/tmp.HldWAbt7ur/target
/tmp/tmp.HldWAbt7ur/target
/tmp/tmp.HldWAbt7ur/target/source
/tmp/tmp.HldWAbt7ur/target/source/b
/tmp/tmp.HldWAbt7ur/target/source/a
/tmp/tmp.HldWAbt7ur/target/b
/tmp/tmp.HldWAbt7ur/target/a
```

Adding `-T` seems to do the right thing.

```
       -T, --no-target-directory
              treat DEST as a normal file
```

```suggestion
        subprocess.check_call(['cp', '-alT', old_extra, new_extra + '.tmp'])
```